### PR TITLE
this commit will add the topic query to the articles endpoint

### PR DESCRIPTION
**Pull request by harrycs1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -78,14 +78,41 @@ describe('/api/articles', () => {
...
+    test('GET:400 responds with an appropriate error if the topic query does not exist', () => {
+        return request(app)
+          .get('/api/articles?topic=banana')
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Topic does not exist");
+          });
+      });
+
+      test('GET:404 responds with an appropriate error if the topic query exists but has no articles', () => {
```
Comment (ghost) [inline __tests__/app.test.js]: Is it an error if the topic exists but doesn't have any associated articles yet? I would argue no.

**Review by ghost — CHANGES_REQUESTED** [review]
Really good work on this, a small change to make. I don't believe one of your 404 should be a 404. If a topic exists that is not a problem if it doesn't have articles. That should be a 200. If you could look into that and resubmit please.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -78,14 +78,41 @@ describe('/api/articles', () => {
...
+            expect(body.msg).toBe("Topic does not exist");
+          });
+      });
+
+      test('GET:200 responds with an appropriate message if the topic query exists but has no articles', () => {
+        return request(app)
+          .get('/api/articles?topic=paper')
+          .expect(200)
+          .then(({ body }) => {
+            expect(body.articles[0]).toBe("No results found");
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: you would just send back an empty array. You'll break the frontend with this at the moment as it expects objects in the array, and you've passed a string. 

Just pass an empty array and the front end can decide what message to show

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
for your last 200 test - you would just send back an empty array. You'll break the frontend with this at the moment as it expects objects in the array, and you've passed a string. 

Just pass an empty array and the front end can decide what message to show

**Pull request merged**